### PR TITLE
Update to xen 4.14 and go 1.16

### PR DESCRIPTION
**Pull request by cavus700**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM fedora:34
```
Comment (diconico07) [inline Dockerfile:1]: I'd rather use an official golang image to benefit from golang version in tag (like eg `golang:1.16-alpine3.13`, I chose an alpine 3.13 one here as it also have xen 4.14 in its repository, but I'm not allergic to other flavors :smile: )

**Review by diconico07 — CHANGES_REQUESTED** [review]
First, thanks for the quite complete PR,
I however see many changes unrelated to go1.16 and xen4.14 upgrade, can you split this lone commit in multiple commits for every change, I see those at least (I let you order them as you want):
 - Docker image building
 - Upgrade to go 1.16
 - Upgrade to Xen 4.14
 - Split into two packages

Except the need to split in multiple commits and my slight issue with the choice of docker's base image, it looks good to me.

File: Dockerfile (hunk 1 above)
Comment (cavus700) [reply Dockerfile:1]: Thank you for the interest in my PR :) I split the commits in three. It was not so easy to make seperate commits out of the go and xen update so I hope it is ok to have only one for it.
I would also like to use an golang/alpine container but I could not make it compile the project. I always got some cgo errors related to xen files. Maybe I just missed an apk package or something is different there with the xen-dev package.

Comment (diconico07) [reply Dockerfile:1]: Ok, fine with me then as it seems there is no official golang image that fits.

**Review by diconico07 — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module github.com/xenlight_exporter
```
Comment (nlewo) [inline go.mod]:
```suggestion
module github.com/gandi/xenlight_exporter
```

File: main.go [hunk 3]
```diff
@@ -2,18 +2,29 @@ package main
...
+	"net/http"
+
+	"github.com/go-kit/log/level"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
-	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/promlog"
+	"github.com/prometheus/common/promlog/flag"
 	"github.com/prometheus/common/version"
+	collect "github.com/xenlight_exporter/pkg/collectors"
```
Comment (nlewo) [inline main.go]:
```suggestion
	collect "github.com/gandi/xenlight_exporter/pkg/collectors"
```

**Review by nlewo — CHANGES_REQUESTED** [review]

**Review by nlewo — APPROVED** [review]
Thanks @cavus700 !

**Pull request merged**